### PR TITLE
Update RT10xx clock init

### PR DESCRIPTION
**Pull request by mmahadevan108**
The clock initialization code is moved out of the soc.c file to the boards folder. The MCUXpresso Config Tool was used to generate clock init code. This change allows customizing the clock initialization code for each MXRT10xx board. 
A MCUXpresso Config Tool file is also provided, this can be used a s starting point when updating the clock configuration.

`clock_config.c` was generated by the tool. This matches what was originally present inside `soc_rt10xx.c`.

### CODE REVIEW

**Comment by hakehuang** [comment]
all test pass, no issue found

**Review by hakehuang — APPROVED** [review]

**Review by Kludentwo — APPROVED** [review]
Looks good to me.

**Comment by carlescufi** [comment]
@mmahadevan108 please replace the PR reference in `west.yml` with a SHA

File: soc/arm/nxp_imx/rt/soc_rt10xx.c [hunk 1]
```diff
@@ -14,75 +14,25 @@
 #include <arch/cpu.h>
 #include <arch/arm/aarch32/cortex_m/cmsis.h>
 #include <fsl_flexspi_nor_boot.h>
+#include <clock_config.h>
```
Comment (mniestroj) [inline soc/arm/nxp_imx/rt/soc_rt10xx.c:17]: This assumes that there is `clock_config.h` already provided. There are several issues with it:
- at the point when this commit is introduced, there are no board specific `clock_config.h` files (they are introduced in subsequent commits), so this change breaks bisectability of the codebase,
- this change enforces new platforms (boards) to always come with generated clock configuration, which is additional level besides device-tree to configure; not sure that is desirable,
- including board specific file from SoC code seems a bit hacky; I think that having a default configuration for SoC is much easier for introducing new boards in the future, while providing an option for board/application specific fine-tuned clock configuration.

File: boards/arm/mimxrt1010_evk/mimxrt1010_evk.mex [hunk 2]
```diff
@@ -0,0 +1,1060 @@
+<?xml version="1.0" encoding= "UTF-8" ?>
+<configuration name="MIMXRT1011xxxxx" xsi:schemaLocation="http://mcuxpresso.nxp.com/XSD/mex_configuration_10 http://mcuxpresso.nxp.com/XSD/mex_configuration_10.xsd" uuid="de3bdbe5-65cf-482d-b1be-cd17b780459b" version="10" xmlns="http://mcuxpresso.nxp.com/XSD/mex_configuration_10" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
```
Comment (mniestroj) [inline boards/arm/mimxrt1010_evk/mimxrt1010_evk.mex:2]: IMO adding those files should be part of commits that include `clock_config.h` (which is generated from those `.mex` files).

**Review by mniestroj — CHANGES_REQUESTED** [review]
While I see value in being able to quickly configure clock tree using graphical configuration tool (e.g. when testing different configurations, taking power consumption measurements, verifying peripherals functionalities), I am not in favor of enforcing it. First of all, this moves us one step farther from device-tree, instead of one step closer. The reason is that after enabling/disabling a peripheral in device-tree (or configuring a specific clock), we will need to regenerate clock configuration anyway. Next thing is that this PR assumes that clock configuration will be platform (board) specific, while clock configuration will be both platform and application specific in lots of cases (e.g. when just a subset of clocks need to be enabled due to specific power consumption optimizations, dependent on board AND application). Lastly, this generated static clock configuration might be harder to combine with runtime clock configuration needed to be handled manually (written by hand, not generated) anyway.

In summary, I like the idea of being able to configure clock tree using MCUxpresson config tool,  but would like to keep it optional, just for subset of use cases (e.g. when enabling a Kconfig option that will instruct build system to use generated clock config).

**Comment by henrikbrixandersen** [comment]
> While I see value in being able to quickly configure clock tree using graphical configuration tool (e.g. when testing different configurations, taking power consumption measurements, verifying peripherals functionalities), I am not in favor of enforcing it. First of all, this moves us one step farther from device-tree, instead of one step closer. The reason is that after enabling/disabling a peripheral in device-tree (or configuring a specific clock), we will need to regenerate clock configuration anyway. Next thing is that this PR assumes that clock configuration will be platform (board) specific, while clock configuration will be both platform and application specific in lots of cases (e.g. when just a subset of clocks need to be enabled due to specific power consumption optimizations, dependent on board AND application). Lastly, this generated static clock configuration might be harder to combine with runtime clock configuration needed to be handled manually (written by hand, not generated) anyway.

I agree.

**Comment by mmahadevan108** [comment]
@mniestroj Thank you for your review, responses below:

> While I see value in being able to quickly configure clock tree using graphical configuration tool (e.g. when testing different configurations, taking power consumption measurements, verifying peripherals functionalities), I am not in favor of enforcing it. First of all, this moves us one step farther from device-tree, instead of one step closer. The reason is that after enabling/disabling a peripheral in device-tree (or configuring a specific clock), we will need to regenerate clock configuration anyway. 
We are not enforcing use of the clock tool. The `clock_config.c` is the output of the tool, we have included the `.mex` configuration file as a starting point in case anyone wish to make clock changes using the tool.  However it is not required to use the tool to have a clock_config.c file.
One of the big benefits of using the tool is it helps identify clock configuration errors (something we spotted in our Zephyr clock init code).
> Next thing is that this PR assumes that clock configuration will be platform (board) specific, while clock configuration will be both platform and application specific in lots of cases (e.g. when just a subset of clocks need to be enabled due to specific power consumption optimizations, dependent on board AND application). Lastly, this generated static clock configuration might be harder to combine with runtime clock configuration needed to be handled manually (written by hand, not generated) anyway.

> In summary, I like the idea of being able to configure clock tree using MCUxpresson config tool, but would like to keep it optional, just for subset of use cases (e.g. when enabling a Kconfig option that will instruct build system to use generated clock config).

**Comment by mmahadevan108** [comment]
@mniestroj Thank you for your review, responses below:
> While I see value in being able to quickly configure clock tree using graphical configuration tool (e.g. when testing different configurations, taking power consumption measurements, verifying peripherals functionalities), I am not in favor of enforcing it. First of all, this moves us one step farther from device-tree, instead of one step closer. The reason is that after enabling/disabling a peripheral in device-tree (or configuring a specific clock), we will need to regenerate clock configuration anyway. 

The `clock_config.c` was generated using the tool. We have included the `.mex` so that users have a starting point when using the tool. However it is not required to use the tool when creating the `clock_config.c` file, we are not enforcing use of the tool.
One of the big benefits of using the tool is that it helps identify errors in the clock setup. 
We do have plans to eventually update the driver clock setup code to use device tree. 

> Next thing is that this PR assumes that clock configuration will be platform (board) specific, while clock configuration will be both platform and application specific in lots of cases (e.g. when just a subset of clocks need to be enabled due to specific power consumption optimizations, dependent on board AND application). 

This is something we have observed in NXP SDK code, clock configuration is board/application specific.

**Comment by gmarull** [comment]
I think that what has done ST for clock config (STM32H7 is a comparable MCU) makes more sense (DT based, with tests, etc.). See for example https://github.com/zephyrproject-rtos/zephyr/pull/41391

**Comment by dleach02** [comment]
@gmarull, we are investigating DTS directions for the longer term but we will still want to leverage this .mex file on the tools provided by our SDK. 

A key aspect of leveraging the .mex and the NXP SDK tools here is that these tools provide a significant level of error/boundaries checking per target board the clocks are being configured for. We are trying to leverage that capability for users who are using our boards or creating custom boards using our silicon. 

So while tests are useful, the collateral provided in our SDK automatically highlights errors in configurations the user attempts to setup.

File: soc/arm/nxp_imx/rt/soc_rt10xx.c (hunk 1 above)
Comment (dleach02) [reply soc/arm/nxp_imx/rt/soc_rt10xx.c:17]: @mmahadevan108, will need to arrange the commits to ensure that the clock_config.h exists before introducing it in this file for the bisectability.

Comment (dleach02) [reply soc/arm/nxp_imx/rt/soc_rt10xx.c:17]: > * this change enforces new platforms (boards) to always come with generated clock configuration, which is additional level besides device-tree to configure; not sure that is desirable,

Currently, the clock configuration code are the lines being deleted from this file. The way it has previously been defined in this file was very error prone.

**Comment by gmarull** [comment]
> @gmarull, we are investigating DTS directions for the longer term but we will still want to leverage this .mex file on the tools provided by our SDK.
> 
> A key aspect of leveraging the .mex and the NXP SDK tools here is that these tools provide a significant level of error/boundaries checking per target board the clocks are being configured for. We are trying to leverage that capability for users who are using our boards or creating custom boards using our silicon.
> 
> So while tests are useful, the collateral provided in our SDK automatically highlights errors in configurations the user attempts to setup.

I agree this sort of tools can help a lot on detecting issues when the system is complex, no objections on that. A path forward would be to take the "mex" file and autogenerate a DT file. A similar procedure is done for [ST pinctrl](https://github.com/zephyrproject-rtos/hal_stm32/tree/main/scripts/genpinctrl): their database is used to generate correct pin definitions in DT files.

**Comment by dleach02** [comment]
> I agree this sort of tools can help a lot on detecting issues when the system is complex, no objections on that. A path forward would be to take the "mex" file and autogenerate a DT file. A similar procedure is done for [ST pinctrl](https://github.com/zephyrproject-rtos/hal_stm32/tree/main/scripts/genpinctrl?rgh-link-date=2022-01-05T14%3A49%3A13Z): their database is used to generate correct pin definitions in DT files.

This is what we are investigating now. We consider this PR a step in that direction while we work to provide fuller enablement of NXP platforms.

File: soc/arm/nxp_imx/rt/soc_rt10xx.c (hunk 1 above)
Comment (mmahadevan108) [reply soc/arm/nxp_imx/rt/soc_rt10xx.c:17]: I have rearranged the commits to address the issue with `git bisect`

**Review by erwango — CHANGES_REQUESTED** [review]
I'm nack'ing this change until it could be reviewed in dev-review meeting.

Here are the main concerns:
- Device tree should be the primary configuration point (cf [Device Tree design goals](https://docs.zephyrproject.org/latest/guides/dts/design.html)), and this change excludes it .I reckon it is mentioned as a long term target, but yet it is excluded today (and long term target is too vague)
- It is a deviation from [board porting general recommendations]:(https://docs.zephyrproject.org/latest/guides/porting/board_porting.html#general-recommendations).
- Question about .mex file upstream in tree per board (is that scalable if applied to all boards? is there any license issue ?)

I don't claim these are blocking points that prevent this PR for being merged. But at least these points are sufficiently important to take the time to discuss about it. Also, if accepted, it is that close to "the limits" that it requires to redefine these limits in order to avoid further changes that could go beyond what is today considered as acceptable. (in this part I include the fact that this change is putting code in tree that is directly generated by an external tool, potentially not open source).

**Comment by dleach02** [comment]
I think people are not quite understanding what we are doing here. 

No one is required to use the configuration tools that ship with the NXP sdk. 

What we are including is an input file our SDK ships so that folks creating solutions with our chips can, if they want to, use the configuration tools to generate the correct clock tree values for their particular board and usage. In Zephyr, for NXP based solutions, the previous way of configuring the clock information was with manual code in the soc.c file. Someone plugged in the values in one of the boards into the NXP tool and realized the values were incorrect. So we wanted to provide a method for folks using NXP parts to leverage the configuration tools to generate correct clock values. As you can see, we also include the reference generated clock tree for the target board just as before. 

> Device tree should be the primary configuration point (cf Device Tree design goals), and this change excludes it .I reckon it is mentioned as a long term target, but yet it is excluded today (and long term target is too vague)

We also agree that ultimately the device tree is the way to go for this. We are internally investigating how we can leverage the collateral we have to automatically generate device tree and to do all the error checking of the parameters being selected.

> It is a deviation from [board porting general recommendations]:(docs.zephyrproject.org/latest/guides/porting/board_porting.html#general-recommendations).

I'm not sure which bullet item(s) we are deviating from?

> Question about .mex file upstream in tree per board (is that scalable if applied to all boards? is there any license issue ?)

The .mex file has no license issues. If there is a concern we can apply one to it to clarify. If we are worried about scaling we can consider other options for directing folks to the relevant .mex file up to and including just asking them to download the specific SDK and extracting it out of there (we don't really want to do that). The file is not directly needed to build Zephyr applications and is only being provided as reference.

File: boards/arm/mimxrt1010_evk/doc/index.rst [hunk 3]
```diff
@@ -191,6 +191,12 @@ see the following message in the terminal:
 
     Hello World! mimxrt1010_evk
 
+MCUXpresso Config Tool
+======================
+
+A ``mimxrt1010_evk.mex`` file is included. This file was used to generate the clock
+initialization code and can be used as a starting point to tweak the clock configuration.
+This could be useful for different boards that are based on i.MX RT1010
```
Comment (erwango) [inline boards/arm/mimxrt1010_evk/doc/index.rst:199]: I would rather expect a "Clock start up configuration" chapter which explains what is the clock configuration provided (all clocks off, if not which ones are active, ...).
In that chapter, you can explain how this was generated, why it is fine vs Zephyr policy and next steps (conversion to device tree).

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/mimxrt1010_evk/doc/index.rst (hunk 3 above)
Comment (mbolivar-nordic) [reply boards/arm/mimxrt1010_evk/doc/index.rst:199]: > vs Zephyr policy and next steps (conversion to device tree).

I think I agree: if it is explicitly documented that this "should" be in devicetree and that you're working towards that, I won't block either.

@dleach02 's can you confirm that the current code in the tree is:

1. wrong,
2. needs to be made board-specific in some way

and

3. NXP's goal is to convert it to a common set of devicetree-based driver(s) that can get the board-specific clock configuration from the board .dts files?

I think that's what you were saying on Discord; I just want to confirm.

Comment (dleach02) [reply boards/arm/mimxrt1010_evk/doc/index.rst:199]: @mmahadevan108, lets update the documentation and also move the .mex file to the HAL (or document how a user can get the file from the NXP sdk).

Comment (dleach02) [reply boards/arm/mimxrt1010_evk/doc/index.rst:199]: @mbolivar-nordic, 

1. Yes. We think when the boards were initially added to the Zephyr project that it copied the wrong values from the SDK. Further, once we started plugging in the values into our configuration tool we found additional error in a clock value that was out of spec.
2. the configuration values for the clock stuff needs to be at the board level. Consider that the crystal frequency selection is a board specific property which impacts the values being used. This will be replaced with a devicetree in the board folder with the specific values needed. 
3. Our goal is to transition all of our in-tree supported boards to device tree.

Comment (dleach02) [reply boards/arm/mimxrt1010_evk/doc/index.rst:199]: @mbolivar-nordic 

(Trying this reply again)

1. Yes, the original code is wrong. We think that maybe the original code copied over from the SDK was incorrect. We further noted that our configuration tool highlighted an incorrect value that was out of specification.
2. Needs to be board specific. Consider that the clock frequency selection is a board specific property that impacts the values being used by the clock configuration. This will eventually be replaced by a board specific devicetree with these values.
3. Yes, we will transition dts

Comment (erwango) [reply boards/arm/mimxrt1010_evk/doc/index.rst:199]: Also, what would be nice is to add a reference to a tracking issue.

Comment (mmahadevan108) [reply boards/arm/mimxrt1010_evk/doc/index.rst]: I have updated the documentation to include a table with the clock configuration on platform initialization. Also a note has been added regarding future device tree support.

File: boards/arm/mimxrt1010_evk/doc/index.rst [hunk 4]
```diff
@@ -191,6 +191,107 @@ see the following message in the terminal:
...
+A ``mimxrt1010_evk.mex`` file is included. This file was used to generate the clock
+initialization code and can be used as a starting point to tweak the clock configuration.
+This could be useful for different boards that are based on i.MX RT1010.
+
+NOTE: The MCUXpresso Config Tool currently generates a ``.c`` file with the clock configuration.
+Considering options on leveraging this tool in the future to generate a devicetree compatible file.
+
+Below is the clock configuration at platform initialization.
+
+- On-chip 24MHz oscillator is enabled
```
Comment (erwango) [inline boards/arm/mimxrt1010_evk/doc/index.rst:206]: As requested in my review, I rather expect a "Clock start up configuration" chapter.

This is not that I don't want MCUXpresso to be mentioned or highlighted, but the main point here is to achieve a correct clock configuration, and that what we expect to be detailed following this change. MCUXpresso is only a way to achieve that (which will soon be replaced by device tree), and, from what you exposed during dev-review meeting, there are no plans to extend use of this tool to generate more code, so the code generation dedicated part would go away at one point, but a "Clock start up configuration" chapter would remain.

File: boards/arm/mimxrt1010_evk/doc/index.rst [hunk 5]
```diff
@@ -191,6 +191,107 @@ see the following message in the terminal:
...
+| SAI3 MCLK 1                | 41.53 MHz       |
++----------------------------+-----------------+
+| SAI3 MCLK 2                | Inactive        |
++----------------------------+-----------------+
+| SAI3 MCLK 3                | 30 MHz          |
++----------------------------+-----------------+
+| SPDIF0_EXTCLK              | Inactive        |
++----------------------------+-----------------+
+| MQS MCLK                   | 41.53 MHz       |
++----------------------------+-----------------+
```
Comment (erwango) [inline boards/arm/mimxrt1010_evk/doc/index.rst:294]: Just a comment here:
IIUC, lot of peripheral clocks are enabled, while we'd expect them to be disabled at start up if not used (in line with the default board configuration). This being said, there is no recommendation on this today, so that's a maintainer's call here.

File: boards/arm/mimxrt1010_evk/doc/index.rst [hunk 6]
```diff
@@ -191,6 +191,107 @@ see the following message in the terminal:
...
 
+MCUXpresso Config Tool
+======================
+
+A ``mimxrt1010_evk.mex`` file is included. This file was used to generate the clock
+initialization code and can be used as a starting point to tweak the clock configuration.
+This could be useful for different boards that are based on i.MX RT1010.
+
+NOTE: The MCUXpresso Config Tool currently generates a ``.c`` file with the clock configuration.
+Considering options on leveraging this tool in the future to generate a devicetree compatible file.
```
Comment (erwango) [inline boards/arm/mimxrt1010_evk/doc/index.rst:202]: My personal feeling is that doesn't sound like a firm commitment, but I won't not block if others are fine with it. @mbolivar-nordic, is that wording ok to you ?

**Review by erwango — DISMISSED** [review]

File: boards/arm/mimxrt1010_evk/doc/index.rst (hunk 5 above)
Comment (mmahadevan108) [reply boards/arm/mimxrt1010_evk/doc/index.rst:294]: The clock is being setup here, it's use is disabled and enabled by the driver whenever needed.

File: boards/arm/mimxrt1010_evk/doc/index.rst (hunk 6 above)
Comment (mmahadevan108) [reply boards/arm/mimxrt1010_evk/doc/index.rst:202]: I am pasting what was mentioned in the DIscord chat. 
"Like on the pinctrl changes, if we want the larger project to completely transition to DTS on the clock control we should put in an RFC that will establish an agreed on timeline to get this done."

File: boards/arm/mimxrt1010_evk/doc/index.rst (hunk 4 above)
Comment (dleach02) [reply boards/arm/mimxrt1010_evk/doc/index.rst:206]: @erwango, do you have any examples you can point us too? 

I'm asking because we have a lot of documentation about clocks in our reference manuals that I don't really want to reproduce here (fairly long) so want to see what others do in this area.

Comment (mmahadevan108) [reply boards/arm/mimxrt1010_evk/doc/index.rst]: I have moved the clock details to a new section.

File: boards/arm/mimxrt1010_evk/doc/index.rst (hunk 6 above)
Comment (erwango) [reply boards/arm/mimxrt1010_evk/doc/index.rst:202]: So, bottom line:
- this statement is purely about tool leveraging. Which, of course, is fine.
- you don't mention any commitment to conversion to DT. You will apply to such conversion when the project as a whole starts a transition activity to DTS for clock control. Which indeed is not worth mentioning.

So there is no mention about plans of any kind for transitioning clocks to DT.
Anyway, I don't think I have the legitimacy to block on such point. I let others comment if needed.
So please consider this remark solved from my perspective.

**Comment by mmahadevan108** [comment]
@mniestroj awaiting your response.

**Comment by henrikbrixandersen** [comment]
How would we go about somebody submitting a PR with changes to one of the boards clock control `.c` file without a corresponding change the included `.mex` file?

**Review by gmarull — CHANGES_REQUESTED** [review]
-1 for the reasons I explain below, but can be dismissed.

- There is a language to describe hardware: Devicetree.
- A clock control API exists. Even if not a _mature_ API, it is being used by other vendors. In the end the value of Zephyr is the Kernel and its ecosystem (e.g. APIs), allowing for a more generic solution. Moving code to board/soc goes in the direction of "Kernel + HAL" (as do other RTOSes).
- Autogenerated C code of such kind should, in general, stay away from the repository. It is hard to review (if possible at all). And in this particular case it does not even follow Zephyr coding conventions (look at `clock_config.c`).
- I think _shortcuts_ should, in general, not be allowed. Zephyr has its own goals/rules, and they should take precedence. Shortcuts may be allowed under certain circumstances, but they should come with a price: for example, for a fixed period of time or some releases. If shortcuts are _free_, what is the incentive to do things better, or to make vendors collaborate on improving/adding APIs?

File: soc/arm/nxp_imx/rt/soc_rt10xx.c (hunk 1 above)
Comment (mniestroj) [reply soc/arm/nxp_imx/rt/soc_rt10xx.c:17]: What was that error-prone previously and has been improved as part of this PR? This is not clear to me.

Having `#include <clock_config.h>` here is a soft requirement to use code generation from MCUxpresso config tool, because that tool actually creates `clock_config.h` file together with implementation of `clock_init()` function (which is part of generated `clock_config.c`. Even though using code generation is not a strict requirement, developer needs to mimic the same behavior (implement `clock_init()` function and provide `clock_config.h` header), which seems weird.

**Review by mniestroj — CHANGES_REQUESTED** [review]
I have used "MCUxpresso config tool" during last two weeks to generate clock initialization code to reduce power consumption of `mimxrt1060_evk` board,. The idea was to have peripherals like UART/CAN/I2C still operating, while keeping MCU in lowest possible power consumption state. Here are my observations:
- Generated `clock_config.h` and `clock_config.c` are not equal to the files provided in this PR, mostly due to whitespace changes and code comments style. I guess that those files were modified by hand in order to comply with Zephyr coding style. Unfortunately this makes it very hard to see what are the generated code differences after regenerating those files after single change in clock tree. IMO generated files should not be modified by hand.
- Generated code from "MCUxpresso config tool" is invalid (does not allow board to boot Zephyr successfully) after disabling PLL1 (ARM PLL) and using directly 24MHz OSC as source clock. This is one of the simplest things developer wants to reconfigure in the system and "MCUxpresso config tool" fails in assisting doing so. Part of the generated code is good, but unfortunately generated code switches back to PLL1 (which is inactive at the time) for a short time, so ARM core is not clocked and code is not executed further.
- i.MX RT MCUs have an embedded DCD table, which also does some clock configuration. This makes initial clock configuration in C even more challanging. Clock configuration depends now on 3 things: default ROM clock configuration, DCD table and clock configuration in C. At each step we need to know what is the starting configuration, so that PLLs can be properly reconfigured with glitches muxes, etc. I am not sure that "MCUxpresso config tool" takes into account what has been used as part of DCD table (how clocks were configured before they are reconfigured once again as part of generated C code).
- Output of "MCUxpresso config tool" was useful mostly to generate code from original and new clock configuration, then using `diff` in order to find out what function calls have changed and copying those changed lines into C code manually. The main reason is previous point about PLL1 - using generated code directly does not allow platform to boot anymore.
- When working with runtime power management, most of the clock reconfiguration needs to be written by hand in C anyway. I doubt that tools like "MCUxpresso config tool" will by smart enough anytime soon to support us in that case. That is another reason why this tool was useful only to generate code differences between one and the other clock tree configuration.
- While using "MCUxpresso config tool" assists developer a lot, using generated code directly has nearly no sense to me: it is bloated, hard to debug, not supporting runtime PM and often not working at all. IMO if there were issues with previous clock configuration code, then it should be fixed (with patches going into 2.7 stable branch, etc.) instead of replaced by generated code (and modified by hand to comply with Zephyr coding style).

Main reason for -1 is that SoC code assumes `clock_config.h` with `clock_init()` implementation provided somewhere. This is generally a requirement to either use generated code from "MCUxpresso config tool" or try to mimic it by implementing `clock_config.h` and `clock_init()` in board support code by hand. This complicates adding support for new imxrt10xx board into Zephyr. In theory clock configuration should not be needed if DCD table would configure everything correctly in the first place.

**Comment by dleach02** [comment]
> How would we go about somebody submitting a PR with changes to one of the boards clock control `.c` file without a corresponding change the included `.mex` file?

We can remove the .mex file if it is problematic. But you are generally correct that the mex changes as someone makes changes to the clock configuration.

**Comment by mmahadevan108** [comment]
I have submitted Issue https://github.com/zephyrproject-rtos/zephyr/issues/41890 to provide details on what is being addressed. I will close the PR and initiate a separate PR with updates to the clock setup without using a tool generated `clock_config` file

**Comment by mmahadevan108** [comment]
I have submitted a new PR https://github.com/zephyrproject-rtos/zephyr/pull/41927 focused on the fixes to the clock initialization code.
